File: ruby/053-etl/etl.rb
class ETL
  # @param [Hash] old_point
  def self.transform(old_system)
    new_system = {}
    old_system.each do |point, chars|
      chars.each { |char| new_system[char.downcase] = point }
    end
    new_system.sort_by { |key, value| key }.to_h
  end
end

File: go/48-etl/etl.go
package etl

import (
	"strings"
)

func Transform(in map[int][]string) map[string]int {
	result := make(map[string]int)
	for point, chars := range in {
		for _, char := range chars {
			result[strings.ToLower(char)] = point
		}
	}

	return result
}
